libs/durability/durability_disk_failure_recovery_libs.py

```diff
@@ -157,8 +161,59 @@ def get_all_nodes_disks(self, master_obj: LogicalNode, worker_obj: list) -> tupl
                 for cvg in resp[1]:
                     disk_list = resp[1][cvg]['data']
                     for disk in disk_list:
-                        unique_disk_list.append(node.hostname + '$' + cvg + '$' + disk)
+                        val = [node.hostname, cvg, disk]
+                        return_dict['disk' + str(cntr)] = val
+                        cntr += 1
             except KeyError as err:
                 LOGGER.error("Exception while retrieving disk details : %s", err)
                 return False, err
-        return True, unique_disk_list
+        return True, return_dict
+
+    def fail_disk(self, disk_fail_cnt: int, master_obj: LogicalNode,
+                  worker_obj: list, pod_name: str, on_diff_cvg: bool = False) -> tuple:
+        """
+        Return the unique list of failed disks.
+        :param disk_fail_cnt: Number of disks to be failed
+        :param master_obj: Node Object of Master
+        :param worker_obj: list of worker node object
+        :param pod_name: name of the pod
+        :param on_diff_cvg: selects disks from different cvg if set True
+        :return : tuple(bool,dict)
+                 dict of format
+                 {'disk1': ['ssc-vm-g4-rhev4-1059.colo.seagate.com', 'cvg-01', '/dev/sdh'],
+                  'disk2': ['ssc-vm-g4-rhev4-1059.colo.seagate.com', 'cvg-02', '/dev/sdd']}
+        """
+
+        LOGGER.info("No of disks to be failed: %s", disk_fail_cnt)
+        failed_disks_dict = {}
+        resp = self.get_all_nodes_disks(master_obj, worker_obj)
+        if not resp[0]:
+            return resp
+        all_disks = resp[1]
+        LOGGER.info("list of all disks: %s", all_disks)
+
+        if on_diff_cvg:
+            one_disk_per_cvg_dict = {}
+            for disk in all_disks:
+                flg = True
+                for val in one_disk_per_cvg_dict:
+                    if all_disks[disk][0] in one_disk_per_cvg_dict[val] and \
+                            all_disks[disk][1] in one_disk_per_cvg_dict[val]:
+                        flg = False
+                        break
+                if flg:
+                    one_disk_per_cvg_dict[disk] = all_disks[disk]
+            if len(one_disk_per_cvg_dict) < disk_fail_cnt:
+                return False, "Number of cvg are less than disk fail count"
```

Comment:
In this function we are only selecting disks to be failed right?
So can't we select number of disks equal to disk_fail_cnt?

Reply:
yes we are selecting disks equal to sidk_fail_cnt only.
this check is only for if we select on_diff_cvg as true (disks will be selected from different cvg grp)

Reply:
So if we have 2 cvgs per node and disk_fail_cnt comes out to be 3 (this count is random i believe and depends on parity), test will fail. I am not sure if it will find actual product issue. This failure will be more dependent on how we do our setup for cvgs.
As per my understanding, it will be better if we select one disk from each cvg and not have this check of len(one_disk_per_cvg_dict) < disk_fail_cnt. 
Please correct if I am wrong.

Reply:
we can not select one disk per cvg as it is, because we have to make sure number of disks to be failed should not be more than K(parity units). so we came up with two scenarios 
1. where we are selecting fail count (randomly) less than K and on different cvg
2. fail count equal to K on different cvg
One thing we should change is instead of failing reduce the fail count and check again

Reply:
changes are done